### PR TITLE
Revocations

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: lib/http-delegations.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    asyncHandler(async (req, res) => {
+      // TODO: consider invoking the `account` as a zcap here ... and add
+      // `invoker` and `delegator` to the account to avoid having a permission
+      // based model here; automatically add an admin account as an invoker
+      // whenever the account is retrieved that can't be removed
+      const {actor} = req.user;
+      // if controller is not specified it will be the authenticated user
+      const {controller = actor.id, capability, delegator, domain} = req.body;
+      // const referenceId = req.body.referenceId || capability.referenceId;
+      if(actor.id !== controller) {
```
Comment (dlongley) [inline lib/http-delegations.js]: JFYI - Reviewing this is blocked on design considerations around `controller` here and whether not these delegations should instead be stored in an EDV.

Comment (mattcollier) [reply lib/http-delegations.js]: @dlongley have decisions been made?  How does that affect this module/PR?

**Comment by mattcollier** [comment]
@dlongley is this module even a thing now?  I went to take out the `delegations` endpoint and realized that I don't think this is used for anything now?

**Comment by dlongley** [comment]
I think it's still being used (presently) to store zcaps for invokers. This may change in the future, but I think it's in use at present.

**Comment by mattcollier** [comment]
@dlongley ready for another pass here.  Either I've removed too much or this PR doesn't do a whole lot aside from adding a test framework and some refactoring.

EDIT: I have not yet rebased away commits if we need to bring back any of the revocation/delgation stuff.

**Review by dlongley — APPROVED** [review]

**Pull request merged**